decentraland/screens/ListLandForSaleScreen.js

```diff
@@ -4,78 +4,60 @@ import { setLandForSaleList, selectLandToBuy } from "../redux/actions";
 import PropTypes from "prop-types";
 import LandForSaleList from "@presentational/LandForSaleList";
 import LandForSaleListItem from "@presentational/LandForSaleListItem";
+import {
+  prepareParcelForSale,
+  prepareEstateForSale,
+  addressesAreEqual,
+  showError,
+  showInfo,
+  getContracts,
+} from "./helpers";
 
-import ContractsAddresses from "@constants/ContractsAddresses";
-const { ESTATE_ADDRESS, MARKETPLACE_ADDRESS } = ContractsAddresses;
-
-import { Action } from "tasit-sdk";
-const { ERC721, Marketplace } = Action;
-const { Estate } = ERC721;
-const { Decentraland: DecentralandMarketplace } = Marketplace;
+import DecentralandUtils from "tasit-sdk/dist/helpers/DecentralandUtils";
 
 export class ListLandForSaleScreen extends React.Component {
-  estateContract = new Estate(ESTATE_ADDRESS);
-  marketplaceContract = new DecentralandMarketplace(MARKETPLACE_ADDRESS);
-
   componentDidMount = async () => {
     try {
       const { setLandForSaleList } = this.props;
-      const landForSaleList = await this._getLandForSaleList();
+      showInfo("Loading assets for sale...");
+      const landForSaleList = await this._getAssetsForSale();
       setLandForSaleList(landForSaleList);
     } catch (err) {
-      console.error(err);
+      showError(err);
     }
   };
 
-  // Note: This function is assuming that:
-  // - All estates have a sell order
-  // - The total supply of estates is small
-  // TODO: Rewrite this function when we move to testnet
-  _getLandForSaleList = async () => {
-    const orders = [];
-    const totalSupply = await this.estateContract.totalSupply();
-
-    for (let estateId = 1; estateId <= Number(totalSupply); estateId++) {
-      const order = this._getLandForSale(estateId);
-      orders.push(order);
-    }
+  _getAssetsForSale = async () => {
+    const decentralandUtils = new DecentralandUtils();
+    const { getOpenSellOrders } = decentralandUtils;
 
-    if (orders.length == 0) console.warn("There isn't any land for sale right now.");
+    const fromBlock = 0;
+    const openSellOrdersEvents = await getOpenSellOrders(fromBlock);
```

Comment:
Showing placeholder land while this is running would go a long way. Maybe 5 grey blocks with placeholder names?

Reply:
https://github.com/tasitlabs/tasit/issues/150

Reply:
Thanks, could I close this one?

Reply:
Yep, I'll resolve it now.